src/test/kotlin/no/nav/syfo/cronjob/virksomhetsnavn/PersonOppfolgingstilfelleVirksomhetsnavnCronjobSpek.kt

```diff
@@ -0,0 +1,338 @@
+package no.nav.syfo.cronjob.virksomhetsnavn
+
+import io.ktor.server.testing.*
+import io.ktor.util.*
+import io.mockk.*
+import kotlinx.coroutines.runBlocking
+import no.nav.syfo.application.cache.RedisStore
+import no.nav.syfo.client.azuread.AzureAdClient
+import no.nav.syfo.client.ereg.EregClient
+import no.nav.syfo.domain.PersonIdent
+import no.nav.syfo.oppfolgingstilfelle.kafka.*
+import no.nav.syfo.personstatus.*
+import no.nav.syfo.personstatus.domain.OversikthendelseType
+import no.nav.syfo.testutil.*
+import no.nav.syfo.testutil.UserConstants.ARBEIDSTAKER_FNR
+import no.nav.syfo.testutil.UserConstants.VIRKSOMHETSNUMMER_DEFAULT
+import no.nav.syfo.testutil.UserConstants.VIRKSOMHETSNUMMER_NO_VIRKSOMHETSNAVN
+import no.nav.syfo.testutil.assertion.checkPPersonOppfolgingstilfelleVirksomhet
+import no.nav.syfo.testutil.assertion.checkPPersonOversiktStatusOppfolgingstilfelle
+import no.nav.syfo.testutil.generator.generateKOversikthendelse
+import no.nav.syfo.testutil.generator.generateKafkaOppfolgingstilfellePerson
+import org.amshove.kluent.shouldBeEqualTo
+import org.amshove.kluent.shouldBeNull
+import org.apache.kafka.clients.consumer.*
+import org.apache.kafka.common.TopicPartition
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+import redis.clients.jedis.*
+import java.time.Duration
+
+@InternalAPI
+object PersonOppfolgingstilfelleVirksomhetsnavnCronjobSpek : Spek({
```

Comment:
Test-navnene her sier ikke noe om testing av cronjobben så kanskje ikke denne burde hete CronjobSpek? Evt om man burde splitte opp i ulike tester for kafka-innlesing (som hovedsakeig testes her?) og cronjob.

Reply:
Gikk litt fort her ja, glemte å endre testenavnene :/

Reply:
Ser at jeg ble litt sen her, men jeg limer inn kommentaren min under:

Jeg synes disse testene er veldig vanskelig å skjønne 🤷 Hvorfor er teksten her "lag status hvis det ikke finnes"? Det som faktisk testes her er vel at cronjoben ikke oppdaterer noe hvis det ikke finnes en aktiv oppgave? Det er jo også unødvendig å ha 40 linjer med test som tester ting relatert til opprettelse av tilfelle: Det testes andre steder, og er ikke relevant for cronjoben 🤔

Reply:
Med andre ord: Fint hvis `it`ene får mer beskrivende navn 👍 Jeg er uenig i at vi trenger en "mer komplett løype" for testen. Det skaper mye støy og gjør det vanskeligere å skjønne hva som faktisk testes.
Vi kan godt ha en ende-til-ende-test, men da må den faktisk teste fra én ende til en annen, ikke blande sammen alle innganger til appen i samme test.
Og så kan vi heller ha tester for de ulike funksjonene for seg selv - men da uten at vi copy paster kafka-lesing og -verifisering overalt 🤷

Reply:
Har nå fikset test-navn, fjernet noen assertions som ikke var så relevant for cronjob-test.
I tillegg så har jeg slått sammen til 1 test som tester hver av de 3 `OversikthendelseType` i en for-løkke så det ikke blir så mye duplikat kode.